src/controls/scale.ts

```diff
@@ -70,6 +70,13 @@ export function scalingIsForbidden(
   if (lockY && by === 'y') {
     return true;
   }
+  // object disappears if scaling 0 sized object so forbid
+  if (fabricObject.width === 0 && by == 'x') {
+    return true;
+  }
+  if (fabricObject.height === 0 && by === 'y') {
+    return true;
+  }
```

Comment:
this is better, but there is the case that `by` is both and should be blocked as well
I would just block it all:
```
return fabricObject.width === 0 || fabricObject.height === 0
```

Reply:
I'll keep the if just to make clear that the case is particular but thanks for pointing out the other by case.